terraform/eval_for_each.go

```diff
@@ -168,3 +172,9 @@ func evaluateForEachExpressionValue(expr hcl.Expression, ctx EvalContext, allowU
 }
 
 const errInvalidForEachUnknownDetail = `The "for_each" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this, use the -target argument to first apply only the resources that the for_each depends on.`
+
+// markSafeLengthInt allows calling LengthInt on marked values safely
+func markSafeLengthInt(val cty.Value) int {
+	v, _ := val.UnmarkDeep()
+	return v.LengthInt()
+}
```

Comment:
Not for this PR, but I'm kind of surprised that `LengthInt` can't be called on marked values. Do you think we could adjust it to do so in the future?

Reply:
Hmm. We could investigate it. There's at least a couple of `Unmark`ing in the format code because of `LengthInt` as well, iirc, i honestly took it (`LengthInt()`)
for granted as something that shouldn't change, but perhaps it should.